app/docs/0.14.x/configuration.md

```diff
@@ -104,13 +107,138 @@ $ export KONG_LOG_LEVEL=error
 
 [Back to TOC](#table-of-contents)
 
-### Custom Nginx configuration & embedding Kong
+### Injecting Nginx directives
 
-Tweaking the Nginx configuration is an essential part of setting up your Kong
-instances since it allows you to optimize its performance for your
-infrastructure, or embed Kong in an already running OpenResty instance.
+Tweaking the Nginx configuration of your Kong instances allows you to optimize
+its performance for your infrastructure.
 
-#### Custom Nginx configuration
+When Kong starts, it builds an Nginx configuration file. You can inject custom
+Nginx directives to this file directly via your Kong configuration.
+
+#### Injecting individual Nginx directives
+
+Any entry added to your `kong.conf` file that is prefixed by `nginx_http_`,
+`nginx_proxy_` or `nginx_admin_` will be converted into an equivalent Nginx
+directive by removing the prefix and added to the appropriate section of the
+Nginx configuration:
+
+* Entries prefixed with `nginx_http_` will be injected to the overall `http`
+block directive.
+
+* Entries prefixed with `nginx_proxy_` will be injected to the `server` block
+directive handling Kong's proxy ports.
+
+* Entries prefixed with `nginx_admin_` will be injected to the `server` block
+directive handling Kong's Admin API ports.
+
+For example, if you the following line to your `kong.conf` file:
+
+```
+nginx_proxy_add_header=Header-Name header-value
+```
+
+This will add the following directive to the proxy `server` block of Kong's
+Nginx configuration:
+
+```
+    add_header Header-Name header-value;
+```
+
+Like any other entry in `kong.conf`, these directives can also be specified
+using [environment variables](#environment-variables) as shown above. For
+example, if you declare an environment variable like this:
+
+```
+$ export KONG_NGINX_HTTP_LUA_SHARED_DICT="custom_cache 64k"
```

Comment:
Maybe let's keep shm injection on the side (until we support injecting multiple ones, but mostly because plugins should be able to inject their own in a more "native" fashion). The same example as the one given above (updated to large_client_header_buffers) maybe? Or another "typical" usage, such as another buffer size, or connection-reuse setting?